CHANGELOG.rst

```diff
@@ -5,6 +5,17 @@ Changelog
 TBR
 ---
 
+* Some changes in behavior when the callback isn't specified (i.e. ``None``) in
+  a ``Request`` which defaults to the ``parse()`` method. A ``UserWarning``
+  would be issued if the following are encountered:
+
+    * Fixed the issue where downloading the response is skipped when the
+      ``parse()`` method is annotated with ``scrapy_poet.DummyResponse``.
+      For instance: ``def parse(self, response: DummyResponse)``.
+    * The ``InjectionMiddleware`` won't attempt to build any dependencies anymore
+      when ``Request.callback == None`` and the ``parse()`` method has annotated
+      dependencies that involves any of ``SCRAPY_POET_PROVIDERS``.
```

Comment:
Updated the **CHANGELOG.rst** to mention `RobotsTxtMiddleware`/`ImagesPipeline`/`FilesPipeline`.

For the **documentation**, I created a new **Pitfalls** doc page to house all peculiarities of this package.

For the **warning**, I think we can omit mentioning these three (3) Scrapy components to have the message more concise. Instead, I think we can point the user to the **Pitfalls** docs for a more in-depth explanation.

> For this same reason, I wonder if in code examples we should also use something other than parse, as you did in a prior version of these changes, to minimize the chance of issues when users copy examples.

I think the current set of examples should be fine even if users copy them since we already fixed the incorrect behavior. The main thing we need to update is dependencies not being built as `parse()` needs to be declared as a callback manually in `start_requests()`. This is done.

If the user fails to do this by writing something like the following _(which has now since been removed in our docs/examples)_:

```python
class Spidey(Spider):
    name = "milesmorales"
    start_urls = ["https://dailybugle.com"]

    def parse(self, response: DummyResponse, page: NewPage):
        ...
```

They'd be met with a `TypeError` since `page` isn't injected + a `UserWarning` explaining what happened:

> A request has been encountered with callback=None which defaults to the parse() method. On such cases, annotated dependencies in the parse() method won't be built by scrapy-poet. However, if the request has callback=parse, the annotated dependencies will be built.